autogen/generator.py

```diff
@@ -320,11 +320,16 @@ def __call__(self):
         D = sorted(D.values(), key=lambda d: d['function'])
         sys.stdout.write("Generating PARI functions:")
 
-        self.gen_file = open(self.gen_filename + '.tmp', 'w')
+        # Stupid Python 3 forces us to specify an encoding
+        if "encoding" in open.__doc__:
+            kwds = dict(mode="wt", encoding="utf-8")
+        else:
+            kwds = dict(mode="wt")
```

Comment:
Right. For this to work, I need to do everything in `unicode` though. Right now, I do an explicit encoding `unicode` -> `str` on Python 2: https://github.com/defeo/cypari2/blob/master/autogen/generator.py#L289

So I would need to delete that and then use `from __future__ import unicode_literals` everywhere.